### PR TITLE
Exec

### PR DESCRIPTION
**Pull request by grondinjc**
Console to execute code.
No state with console.

### CODE REVIEW

**Comment by grondinjc** [comment]
@vphilippon 
During the demo, the file did not print ... but it is normal since main() is never called ... 
def f(arg):
  print arg
def main():
  f("allo")

**Comment by grondinjc** [comment]
@vphilippon In my logs, I saw the error set.discard(user) that you fixed...
Those changes were not in my branch during demo.

**Comment by grondinjc** [comment]
Added server response to console action to debug in various contexts
Error messages should be reported in browser console. :+1:

File: src/app/python/core.py [hunk 1]
```diff
@@ -391,6 +443,125 @@ def _task_file_edit(self, path, changes, user):
       self._project_files[path].file.add(bundle)
 
   @task_time(microseconds=1)
+  def _task_create_archive(self, path, caller, response):
```
Comment (vphilippon) [inline src/app/python/core.py:446]: @grondinjc _task_create_archive already exists (line 600).

File: src/server/ideController.py [hunk 2]
```diff
@@ -519,29 +634,149 @@ def _dump_callback(self, result, caller):
...
                                                                       ws.peer_address[0],
                                                                       ws.peer_address[1]))
       except:
-        self._logger.error("{0} ({1}:{2}) WS transfer failed".format(caller,
+        self._logger.error("{0} ({1}:{2}) WS transfer failed".format(user,
                                                                      ws.peer_address[0],
                                                                      ws.peer_address[1]))
         # Close websocket
-        ws.close(reason='Failed to send dump')
+        ws.close(reason='Failed to send exec output')
```
Comment (vphilippon) [inline src/server/ideController.py:776]: The close reason is not always the same ('Failed to send exec output')

File: src/app/python/core.py [hunk 3]
```diff
@@ -104,6 +108,10 @@ def __init__(self, project_conf, core_conf, logger):
     first_strategy = StrategyCallEmpty(self._change_core_strategy)
     self._core_listeners_strategy = first_strategy
 
+    # Client executions
+    # Association user -> Exec(pipestdin, pipestdout)
```
Comment (vphilippon) [inline src/app/python/core.py:112]: Change for:
_#_ Association user -> Exec(process, file, args)

**Comment by vphilippon** [comment]
And with other changes, the PR can't be merged automatically. Will need a merge on your side, I guess.

File: src/app/python/core.py (hunk 1 above)
Comment (grondinjc) [inline src/app/python/core.py:446]: bad conflict merge

**Pull request closed without merging**